codegen/package.go

```diff
@@ -94,8 +95,9 @@ type PackageHelperOptions struct {
 	// relative path to the default middleware config dir, defaults to ""
 	RelDefaultMiddlewareConfigDir string
 
-	// package path to the generated code, defaults to PackageRoot + "/" + RelTargetGenDir + "/gen-code"
-	GenCodePackage string
+	// map of idl type to package path of the generated code
+	// map should be keyed with idl file extension, e.g., .thrift, .proto
+	GenCodePackage map[string]string
```

Comment:
This is fine but probably we when reading from config, we lowercase the ext and store in the map

Reply:
They have to be lowercase when provided by user, otherwise won't be valid file extension. The lowercase is also mandated by the the schema defined in this PR https://github.com/uber/zanzibar/pull/728/files/fa0b790cda65e487f2d88a6c1d34e64ca91b56a8#diff-d750b667b8c67069eee0381c084dc1caR23